### PR TITLE
Update DTH tests to reduce time spent on timeout

### PR DESCRIPTION
**Pull request by troydai**
Further cut back time spent on timeout. The locally running time reduced to 200s from 300s.

/cc @davidfowl 
/review @JunTaoLuo @victorhurdugaci @BrennanConroy

### CODE REVIEW

**Comment by troydai** [comment]
Travis passed. Ping.

File: test/Microsoft.Dnx.DesignTimeHost.FunctionalTests/DthStartupTests.cs [hunk 1]
```diff
@@ -448,7 +448,7 @@ public void AddDepsReturnsReferences(DnxSdk sdk)
 
                 client.Initialize(project.ProjectDirectory);
 
-                var messages = client.DrainAllMessages();
+                var messages = client.DrainMessage(13);
```
Comment (davidfowl) [inline test/Microsoft.Dnx.DesignTimeHost.FunctionalTests/DthStartupTests.cs]: Hmmm where did 13 come from?

File: test/Microsoft.Dnx.DesignTimeHost.FunctionalTests/Infrastructure/DthTestClient.cs [hunk 2]
```diff
@@ -105,6 +105,18 @@ public void SetProtocolVersion(int version)
             SendPayLoad(0, ProtocolManager.NegotiationMessageTypeName, new { Version = version });
         }
 
+        public List<DthMessage> DrainMessage(int count)
+        {
+            var result = new List<DthMessage>();
+            while (count > 0)
```
Comment (davidfowl) [inline test/Microsoft.Dnx.DesignTimeHost.FunctionalTests/Infrastructure/DthTestClient.cs]: for loop not cool enough?

**Comment by davidfowl** [comment]
:shipit:

**Comment by davidfowl** [comment]
Add some comments around the numbers.

File: test/Microsoft.Dnx.DesignTimeHost.FunctionalTests/DthStartupTests.cs (hunk 1 above)
Comment (troydai) [inline test/Microsoft.Dnx.DesignTimeHost.FunctionalTests/DthStartupTests.cs]: I'll add comments in code.

File: test/Microsoft.Dnx.DesignTimeHost.FunctionalTests/Infrastructure/DthTestClient.cs (hunk 2 above)
Comment (troydai) [inline test/Microsoft.Dnx.DesignTimeHost.FunctionalTests/Infrastructure/DthTestClient.cs]: Not even close. `while` is the best.

**Comment by troydai** [comment]
Travis passed, I'll merge in seconds.

**Pull request merged**